testsuite/cluster-test/src/cluster_swarm/cluster_swarm_kube.rs

```diff
@@ -456,7 +458,12 @@ impl ClusterSwarmKube {
         let (p, s): (Pod, Service) = match &instance_config.application_config {
             Validator(validator_config) => (
                 self.validator_spec(
-                    instance_config.validator_group,
+                    instance_config.validator_group.index as u32,
+                    match instance_config.validator_group.twin_index {
```

Comment:
This should call into `InstanceConfig::pod_name` which should be updated to handle twin index.
This `format!("val-{}")` part is fragile, we want to make sure there is only single place in codebase that does this formatting(InstanceConfig::pod_name)